diff --git a/vendor/magento/framework/Setup/SampleData/Executor.php b/vendor/magento/framework/Setup/SampleData/Executor.php
index de60f03..2bb70bb 100644
--- a/vendor/magento/framework/Setup/SampleData/Executor.php
+++ b/vendor/magento/framework/Setup/SampleData/Executor.php
@@ -50,12 +50,13 @@
      */
     public function exec(InstallerInterface $installer)
     {
-        try {
+//        try {
             $this->appState->emulateAreaCode(\Magento\Framework\App\Area::AREA_GLOBAL, [$installer, 'install']);
             $this->state->setInstalled();
-        } catch (\Throwable $e) {
-            $this->state->setError();
-            $this->logger->error('Sample Data error: ' . $e->getMessage());
-        }
+//        } catch (\Throwable $e) {
+//            $this->state->setError();
+//            $this->logger->error('Sample Data error: ' . $e->getMessage());
+//
+//        }
     }
 }
