diff --git a/vendor/magento/module-catalog-data-exporter/Model/Provider/Product/Formatter/ScopeFormatter.php b/vendor/magento/module-catalog-data-exporter/Model/Provider/Product/Formatter/ScopeFormatter.php
index 68fbe79..ef97927 100644
--- a/vendor/magento/module-catalog-data-exporter/Model/Provider/Product/Formatter/ScopeFormatter.php
+++ b/vendor/magento/module-catalog-data-exporter/Model/Provider/Product/Formatter/ScopeFormatter.php
@@ -53,7 +53,7 @@ class ScopeFormatter implements FormatterInterface
      */
     private function getScopes()
     {
-        if (!$this->scopes) {
+        //if (!$this->scopes) {
             $connection = $this->resourceConnection->getConnection();
             $select = $connection->select()
                 ->from(['s' => $this->getTable('store')], [])
@@ -77,7 +77,7 @@ class ScopeFormatter implements FormatterInterface
                 ->where('s.store_id != 0');

             $this->scopes = $connection->fetchAssoc($select);
-        }
+        //}
         return $this->scopes;
     }

