diff --git a/vendor/magento/module-catalog/Model/Category/Attribute/LayoutUpdateManager.php b/vendor/magento/module-catalog/Model/Category/Attribute/LayoutUpdateManager.php
index f5694a46d3f..25f91c1c641 100644
--- a/vendor/magento/module-catalog/Model/Category/Attribute/LayoutUpdateManager.php
+++ b/vendor/magento/module-catalog/Model/Category/Attribute/LayoutUpdateManager.php
@@ -98,7 +98,7 @@ class LayoutUpdateManager
             array_map(
                 function (string $handle) use ($category) : ?string {
                     preg_match(
-                        '/^catalog\_category\_view\_selectable\_' .$category->getId() .'\_([a-z0-9]+)/i',
+                        '/^catalog\_category\_view\_selectable\_([a-z0-9]+)/i',
                         $handle,
                         $selectable
                     );
diff --git a/vendor/magento/module-catalog/Model/Product/Attribute/LayoutUpdateManager.php b/vendor/magento/module-catalog/Model/Product/Attribute/LayoutUpdateManager.php
index 92ae9895000..2e3893335a2 100644
--- a/vendor/magento/module-catalog/Model/Product/Attribute/LayoutUpdateManager.php
+++ b/vendor/magento/module-catalog/Model/Product/Attribute/LayoutUpdateManager.php
@@ -110,7 +110,7 @@ class LayoutUpdateManager
             array_map(
                 function (string $handle) use ($identifier) : ?string {
                     preg_match(
-                        '/^catalog\_product\_view\_selectable\_' .preg_quote($identifier) .'\_([a-z0-9]+)/i',
+                        '/^catalog\_product\_view\_selectable\_([a-z0-9]+)/i',
                         $handle,
                         $selectable
                     );
diff --git a/vendor/magento/module-cms/Model/Page/CustomLayout/CustomLayoutManager.php b/vendor/magento/module-cms/Model/Page/CustomLayout/CustomLayoutManager.php
index 988bd5b4ac1..a8bdc8b3ea1 100644
--- a/vendor/magento/module-cms/Model/Page/CustomLayout/CustomLayoutManager.php
+++ b/vendor/magento/module-cms/Model/Page/CustomLayout/CustomLayoutManager.php
@@ -121,7 +121,7 @@ class CustomLayoutManager implements CustomLayoutManagerInterface
             array_map(
                 function (string $handle) use ($identifier) : ?string {
                     preg_match(
-                        '/^cms\_page\_view\_selectable\_' .preg_quote($identifier) .'\_([a-z0-9]+)/i',
+                        '/^cms\_page\_view\_selectable\_([a-z0-9]+)/i',
                         $handle,
                         $selectable
                     );
