diff --git a/vendor/magento/module-google-tag-manager/view/frontend/templates/impression.phtml b/vendor/magento/module-google-tag-manager/view/frontend/templates/impression.phtml
index 7ece3754bd..497f7913d4 100644
--- a/vendor/magento/module-google-tag-manager/view/frontend/templates/impression.phtml
+++ b/vendor/magento/module-google-tag-manager/view/frontend/templates/impression.phtml
@@ -12,7 +12,7 @@ $productCollection = $block->getLoadedProductCollection();
 $block->checkCartItems();
 ?>
 
-<?php if (count($productCollection) > 0) { ?>
+<?php if ($productCollection && count($productCollection) > 0) { ?>
     <?php
     $listPosition = 0;
     if (is_object($productCollection) && $productCollection->getCurPage() > 0) {
