diff --git a/vendor/magento/framework/Backup/Db.php b/vendor/magento/framework/Backup/Db.php
index d9585f2..852016e 100644
--- a/vendor/magento/framework/Backup/Db.php
+++ b/vendor/magento/framework/Backup/Db.php
@@ -45,7 +45,7 @@ class Db extends AbstractBackup
         foreach ($file as $statement) {
             $this->getResourceModel()->runCommand($statement);
         }
-        @unlink($source);
+        //@unlink($source);
 
         $this->_lastOperationSucceed = true;
 
