diff --git a/vendor/magento/magento-cloud-components/Model/Cache/InvalidateLogger.php b/vendor/magento/magento-cloud-components/Model/Cache/InvalidateLogger.php
index 6b3e769c..e597f57e 100644
--- a/vendor/magento/magento-cloud-components/Model/Cache/InvalidateLogger.php
+++ b/vendor/magento/magento-cloud-components/Model/Cache/InvalidateLogger.php
@@ -91,7 +91,7 @@ class InvalidateLogger extends \Magento\Framework\Cache\InvalidateLogger
                 }
             }
 
-            if ($needTrace) {
+            if (false && $needTrace) { // khb: never need trace on demo region
                 $invalidateInfo['trace'] = $this->debugTrace->getTrace();
             }
         }
