--- /vendor/magento/module-customer/Model/Visitor.php	(date 1587498163913)
+++ /vendor/magento/module-customer/Model/Visitor.php	(date 1587498163913)
@@ -193,11 +193,11 @@
     public function saveByRequest($observer)
     {
         // prevent saving Visitor for safe methods, e.g. GET request
-        if (($this->skipRequestLogging || $this->requestSafety->isSafeMethod() || $this->isModuleIgnored($observer))
+        /*if (($this->skipRequestLogging || $this->requestSafety->isSafeMethod() || $this->isModuleIgnored($observer))
             && !$this->sessionIdHasChanged()
         ) {
             return $this;
-        }
+        }*/

         try {
             $this->save();
