Index: /vendor/magento/module-customer/Model/Visitor.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- /vendor/magento/module-customer/Model/Visitor.php	(date 1587570929288)
+++ /vendor/magento/module-customer/Model/Visitor.php	(date 1587570929288)
@@ -171,9 +171,9 @@
         $this->setLastVisitAt((new \DateTime())->format(\Magento\Framework\Stdlib\DateTime::DATETIME_PHP_FORMAT));

         // prevent saving Visitor for safe methods, e.g. GET request
-        if ($this->requestSafety->isSafeMethod()) {
+        /*if ($this->requestSafety->isSafeMethod()) {
             return $this;
-        }
+        }*/

         if (!$this->getId()) {
             $this->setSessionId($this->session->getSessionId());
