Index: vendor/magento/module-customer/Model/Visitor.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- /vendor/magento/module-customer/Model/Visitor.php	(date 1587565176915)
+++ /vendor/magento/module-customer/Model/Visitor.php	(date 1587565176915)
@@ -195,11 +195,11 @@
     public function saveByRequest($observer)
     {
         // prevent saving Visitor for safe methods, e.g. GET request
-        if (($this->skipRequestLogging || $this->requestSafety->isSafeMethod() || $this->isModuleIgnored($observer))
+        /*if (($this->skipRequestLogging || $this->requestSafety->isSafeMethod() || $this->isModuleIgnored($observer))
             && !$this->sessionIdHasChanged()
         ) {
             return $this;
-        }
+        }*/

         try {
             if ($this->session->getSessionId() && $this->getSessionId() != $this->session->getSessionId()) {
